javascript-challenges.test.js

```diff
@@ -23,12 +41,57 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const codeWord =(string) => {
+   let arr = string.toLowerCase().split("")
+//    Declare new variable to lowercase all letters and split into array where all letters are seperated by commas
+   let newArr = []
+//    Declare empty array to put values into
+    for (let i=0; i<arr.length; i++) {
+        // for loop to iterate over each letter of arr
+        if (arr[i] === "a"){
+            newArr.push("4")
+        } else if (arr[i] === "e"){
+            newArr.push("3")
+        } else if (arr[i] === "i"){
+            newArr.push("1")
+        } else if (arr[i] === "o"){
+            newArr.push("0")
+        } else {
+            newArr.push(arr[i])
+        }
+        // conditional statements to change vowels to numbers and push new values into newArr
+    }
+
+    let newWord = newArr.slice(1).join("")
+    // new variable to separate first element from rest of newArr and bring that together as a string
+    let firstLetter = newArr[0].toUpperCase()
+    // new variable to capitalize first element on newArr
+    return firstLetter + newWord
+    // return the result of concatenating the capitalized 0th index with remainder of string
+}
+// console.log(codeWord(secretCodeWord3))
+// 3cc3ntr1c
+// console.log(codeWord(secretCodeWord2))
+// G0bbl3dyg00k
+// console.log(codeWord(secretCodeWord1))
+// L4ck4d41s1c4l
+
```

Comment:
Remember that the purpose of the test is to call your function. So these console log lines should not be part of your workflow. Leave your test commented IN and see the output!